packages/backend/src/open_payments/wallet_address/service.test.ts

```diff
@@ -797,3 +801,92 @@ describe('Open Payments Wallet Address Service', (): void => {
     )
   })
 })
+
+describe('Open Payments Wallet Address Service using Cache', (): void => {
+  let deps: IocContract<AppServices>
+  let appContainer: TestContainer
+  let walletAddressService: WalletAddressService
+  let walletAddressCache: CacheDataStore<WalletAddress>
+  let knex: Knex
+
+  beforeAll(async (): Promise<void> => {
+    deps = initIocContainer({
+      ...Config,
+      localCacheDuration: 5_000 // 5-second default.
+    })
+    appContainer = await createTestApp(deps)
+    knex = appContainer.knex
+    walletAddressService = await deps.use('walletAddressService')
+    walletAddressCache = await deps.use('walletAddressCache')
+  })
+
+  afterEach(async (): Promise<void> => {
+    jest.useRealTimers()
+    await truncateTables(knex)
+  })
+
+  afterAll(async (): Promise<void> => {
+    await appContainer.shutdown()
+  })
+
+  describe('Update and Fetch Wallet Address with cache', (): void => {
+    test.each`
+      initialIsActive | status        | expectedIsActive | expectedCallCount
+      ${true}         | ${undefined}  | ${true}          | ${2}
+      ${true}         | ${'INACTIVE'} | ${false}         | ${2}
+      ${false}        | ${'ACTIVE'}   | ${true}          | ${3}
+      ${false}        | ${undefined}  | ${false}         | ${3}
+    `(
+      'Wallet address with initial isActive of $initialIsActive can be updated with $status status and called $expectedCallCount',
+      async ({
+        initialIsActive,
+        status,
+        expectedIsActive,
+        expectedCallCount
+      }): Promise<void> => {
+        const spyCacheSet = jest.spyOn(walletAddressCache, 'set')
+        const walletAddress = await createWalletAddress(deps)
+        expect(spyCacheSet).toHaveBeenCalledTimes(1)
+
+        if (!initialIsActive) {
+          // Only update the database:
+          await walletAddress.$query(knex).patch({ deactivatedAt: new Date() })
+          const fromCacheActive = await walletAddressService.get(
+            walletAddress.id
+          )
+
+          // We don't expect a match here, since the cache and database is out-of-sync:
+          expect(fromCacheActive!.isActive).toEqual(false)
+
+          // Update through the service, will also update the wallet-address cache:
+          await walletAddressService.update({
+            id: walletAddress.id,
+            status: 'INACTIVE'
+          })
+        }
+
+        const updatedWalletAddress = await walletAddressService.update({
+          id: walletAddress.id,
+          status
+        })
+        assert.ok(!isWalletAddressError(updatedWalletAddress))
+        expect(updatedWalletAddress.isActive).toEqual(expectedIsActive)
+
+        // We expect the [set] to be called again with the new data:
+        expect(spyCacheSet).toHaveBeenCalledTimes(expectedCallCount)
+        expect(spyCacheSet).toHaveBeenCalledWith(
+          walletAddress.id,
+          expect.anything()
```

Comment:
we could probably using something like `expect.objectContaining({ id, status })` such that we can verify the few main fields but don't have to go through every single one